### PR TITLE
Set safeTxGas=0 when safeVersion>=1.3.0

### LINKED ISSUE #100: Send 0 for safeTxGas
## Overview
`safeTxGas` is a workaround that was used so RPC providers could estimate correctly the gas used by a Safe transaction. This way we could get a valid `gasLimit` and avoid running out of gas because transaction was estimated lower than the real cost.
Since v1.3.0 this workaround is not necessary any more as now a transaction failing by low gas will revert instead of succeed, and we could simply send a 0 value and let the wallets to estimate the full cost in gas of the transaction.

From the contract code comments:
```
If no safeTxGas and no gasPrice was set (e.g. both are 0),
then the internal tx is required to be successful
This makes it possible to use `estimateGas` without issues,
as it searches for the minimum gas where the tx doesn't revert
```

## Goals
* Send `safeTxGas=0` when sending a transaction from a Safe that is using at least v1.3.0 version of the smart contract
* Export `standardizeSafeTransactionData`

## Requirements
* For v1.3.0 safes, send a `safeTxGas=0`
* For earlier version (esp. v1.1.1), keep estimating and sending the value

### PR DESCRIPTION
**Pull request by germartinez**
## What it solves
Resolves #100

### CODE REVIEW

File: packages/safe-core-sdk/src/utils/transactions/utils.ts [hunk 1]
```diff
@@ -37,16 +38,22 @@ export async function standardizeSafeTransactionData(
...
-      safeContract,
-      ethAdapter,
-      standardizedTxs.to,
-      standardizedTxs.value,
-      standardizedTxs.data,
-      standardizedTxs.operation
-    ))
+  let safeTxGas: number
+  const safeVersion = await safeContract.getVersion()
+  if (semverSatisfies(safeVersion, '>=1.3.0')) {
```
Comment (katspaugh) [inline packages/safe-core-sdk/src/utils/transactions/utils.ts:43]: Maybe put `'>=1.3.0'` in some sort of config, [similar to safe-react](https://github.com/gnosis/safe-react/blob/a2cd631b8f765c10a1fbf55de9290121e5da0e9b/src/logic/safe/utils/safeVersion.ts)?

Comment (iamacook) [inline packages/safe-core-sdk/src/utils/transactions/utils.ts:43]: Maybe it makes sense to create a lookup for the versions?

File: packages/safe-service-client/tests/endpoint.test.ts [hunk 2]
```diff
@@ -33,210 +33,14 @@ describe('Endpoint tests', () => {
...
       const safeTxData: SafeTransactionDataPartial = {
         to: '0xa33d2495760462018275994d85117600bd58221e',
         data: '0x',
         value: '123456789',
         operation: 1,
-        safeTxGas: 123,
+        safeTxGas: 0,
         baseGas: 0,
         gasPrice: 0,
         gasToken: '0x0000000000000000000000000000000000000000',
```
Comment (iamacook) [inline packages/safe-service-client/tests/endpoint.test.ts:46]: Do you use the a `ZERO_ADDRESS` often? Might be worthwhile creating a constant for it.

**Review by iamacook — APPROVED** [review]
Looking good! I just have a couple of suggestions

**Review by DiogoSoaress — COMMENTED** [review]
I would just address the suggestion of using some kind of lookup for the `safe version`

File: packages/safe-service-client/tests/endpoint.test.ts (hunk 2 above)
Comment (germartinez) [reply packages/safe-service-client/tests/endpoint.test.ts:46]: It is only used twice in this test file

**Review by iamacook — APPROVED** [review]
🚀

**Pull request merged**